selfdrive/ui/qt/widgets/cameraview.cc

```diff
@@ -216,6 +216,10 @@ void CameraViewWidget::paintGL() {
 
   if (latest_frame == nullptr) return;
 
+  if (latest_frame_id != draw_frame_id) {
+    LOGW("camera and draw frame ids do not match: %d != %d", latest_frame_id, draw_frame_id);
+  }
```

Comment:
This will spam when modeld dies, or on PC replay, which is always 1 frame behind